rust_src/src/lists.rs

```diff
@@ -31,6 +31,12 @@ impl LispObject {
         }
     }
 
```

Comment:
Why is this function defined?

Reply:
It was used in the cons reversing implementation that didn't use iterators. It replaces the C macro `CHECK_LIST_END`

Reply:
Ah, I see. This is currently part of the iterator code. Does it need to be refactored out into a stand alone function?

Reply:
I guess it's only necessary if there's places where the iterators can't be used to implement a thing that needs to check if something is the end of a list, which I'm not sure there is, but I could be wrong